user-guide/Advanced_Functionality/Security/Advanced_security_configuration/Security_NATS.md

```diff
@@ -8,100 +8,128 @@ By default, NATS does **not** employ TLS encryption, leaving communication susce
 
 ## Enabling NATS inter-node TLS communication
 
+To enable NATS inter-node TLS communication, first [request or generate the TLS certificates](#request-or-generate-the-tls-certificates-pem-format), and then [configure the TLS certificates](#configure-the-tls-certificates).
+
 > [!NOTE]
 > This applies solely to instances involving a DMS cluster or external DxMs. If you have a single-Agent setup, all communication is confined to the local host, as there is only one NATS node.
 
-1. Request or generate TLS certificates (in PEM format).
+### Request or generate the TLS certificates (PEM format)
+
+To generate self-signed certificates, we recommend that you **use our [scripts for generating TLS certificates](https://github.com/SkylineCommunications/generate-tls-certificates)**, available on GitHub. There is a version of the script for Linux and for Windows machines. The script requires two tools: *openssl* and the *Java keytool*. Both of these can run on Linux and Windows.
 
-   To generate self-signed certificates, we recommend that you **use our [scripts for generating TLS certificates](https://github.com/SkylineCommunications/generate-tls-certificates)**, available on GitHub. There is a version of the script for Linux and for Windows machines. The script requires two tools: *openssl* and the *Java keytool*. Both of these can run on Linux and Windows.
+### Configure the TLS certificates
 
-1. Follow the steps below **for each of the NATS nodes** in the cluster to enable TLS encryption:
+A **NATS node** is always configured on the system where the DataMiner software is hosted. Follow the steps below for **each of the NATS nodes** in the cluster to enable inter-node TLS encryption:
 
-   1. Update the **nats-server.config** file to incorporate the TLS section within the cluster segment.
+1. Ensure the **NATSForceManualConfig** option is enabled (see [Disabling automatic NATS configuration](xref:SLNetClientTest_disabling_automatic_nats_config)).
```

Comment:
@SeppeDejonckheere, could you confirm that this step can be added (also on line 107 below)? Based on the caution under [Disabling automatic NATS configuration](https://docs.dataminer.services/user-guide/Reference/DataMiner_Tools/SLNetClientTest_tool/SLNetClientTest_tool_advanced_procedures/SLNetClientTest_disabling_automatic_nats_config.html), that means users will need to manually maintain several files and ensure the synchronization of credentials in the system. If the step is indeed OK to add, I think it might be best if we also added a caution block at the top of the current page to indicate that this procedure has those consequences.

Reply:
Since the TLS aspect of the configuration isn't done automatically by DataMiner at this point and DataMiner has write control over these config files, it will overwrite them back to the original settings (removing the custom TLS configuration).
At this point in time, it is indeed a choice between automatic configuration and the security aspect of configuring TLS.